org-policy/user_permissions.rego

```diff
@@ -0,0 +1,26 @@
+package diamond
+
+import rego.v1
+
+profile := http.send({
+	"url": opa.runtime().env.PROFILE_ENDPOINT,
+	"method": "GET",
+	"headers": {"authorization": input.attributes.request.http.headers.authorization},
+})
+
+subject := profile.sub
+
+# METADATA
+# entrypoint: true
+user_on_session(proposalNumber, visitNumber) if {
+	some allowed_session in data.diamond.data.users.sessions[subject]
+	allowed_session[0] = proposalNumber
+	allowed_session[1] = visitNumber
+}
+
+# METADATA
+# entrypoint: true
+user_on_proposal(proposalNumber) if {
+	some allowed_proposal in data.diamond.data.users.proposals[subject]
+	allowed_proposal = proposalNumber
```

Comment:
There is a good chance this is just me not knowing rego, but could this be simplified as?

```suggestion
	proposalNumber in data.diamond.data.users.proposals[subject]
```